modules/twinkleimage.js

```diff
@@ -9,15 +9,12 @@
  *** twinkleimage.js: Image CSD module
  ****************************************
  * Mode of invocation:     Tab ("DI")
- * Active on:              File pages with a corresponding file which is local (not on Commons)
+ * Active on:              Local nonredirect file pages (not on Commons)
  * Config directives in:   TwinkleConfig
  */
 
 Twinkle.image = function twinkleimage() {
-	if (mw.config.get('wgNamespaceNumber') === 6 &&
-			!document.getElementById("mw-sharedupload") &&
-			document.getElementById("mw-imagepage-section-filehistory")) {
-
+	if (mw.config.get('wgNamespaceNumber') === 6 && mw.config.get('wgArticleId') && !document.getElementById("mw-sharedupload") && !Morebits.wiki.isPageRedirect()) {
```

Comment:
The condition `!document.getElementById("mw-sharedupload")` seems redundant to  `mw.config.get('wgArticleId')`, as for Commons files, the latter value is 0.

Reply:
A local page can still exist, though, and there are more appropriate options (e.g. F2) than the di menu.